bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/discovery/KNXnetDiscoveryService.java

```diff
@@ -0,0 +1,152 @@
+        try {
+            logger.debug("Starting KNXnet/IP discovery scan");
+            Thread.sleep(1000);
+            Discoverer discovererUdp = new Discoverer(0, false);
+            discovererUdp.startSearch(3, true);
+
+            List<Result<SearchResponse>> responses = discovererUdp.getSearchResponses();
+
+            for (Result<SearchResponse> r : responses) {
+                SearchResponse response = r.getResponse();
+                Map<ServiceFamily, Integer> services = response.getServiceFamilies().families();
+
+                if (services.containsKey(ServiceFamiliesDIB.ServiceFamily.Tunneling)
+                        || services.containsKey(ServiceFamiliesDIB.ServiceFamily.Routing)) {
+                    String serial = Objects.toString(response.getDevice().serialNumber()).replace(':', '-');
+
+                    if (logger.isTraceEnabled()) {
+                        logger.trace("Discovered device {}", response);
+                    } else {
+                        logger.debug("Discovered device {}, {}, {}", response.getDevice().getName(), serial,
+                                response.getDevice().getMACAddressString());
+                    }
+
+                    if (services.containsKey(ServiceFamiliesDIB.ServiceFamily.Tunneling)) {
+                        thingDiscovered(DiscoveryResultBuilder.create(new ThingUID(THING_TYPE_IP_BRIDGE, serial))
+                                .withLabel(response.getDevice().getName()).withProperty("serialNumber", serial)
+                                .withProperty("type", "TUNNEL")
+                                .withProperty("ipAddress",
+                                        "" + response.getControlEndpoint().getAddress().getHostAddress())
+                                .withProperty("port", "" + response.getControlEndpoint().getPort())
+                                .withRepresentationProperty("serialNumber").build());
+                    }
+                    if (services.containsKey(ServiceFamiliesDIB.ServiceFamily.Routing)) {
+                        thingDiscovered(DiscoveryResultBuilder.create(new ThingUID(THING_TYPE_IP_BRIDGE, serial))
+                                .withLabel(response.getDevice().getName() + " (router mode)")
+                                .withProperty("serialNumber", serial + "-r").withProperty("type", "ROUTER")
+                                .withProperty("ipAddress", "224.0.23.12")
+                                .withProperty("port", "" + response.getControlEndpoint().getPort())
+                                .withRepresentationProperty("serialNumber").build());
+                    }
+                } else {
+                    logger.trace("Ignoring device {}", response);
+                }
+            }
+            logger.debug("Completed KNXnet/IP discovery scan");
+        } catch (Exception ex) {
+            logger.info("An error occurred during KNXnet/IP discovery {}", ex.getMessage(), ex);
```

Comment:
I guess this could rather be a warning as it is an unexpected situation, isn't it?

Reply:
Maybe. I consider this feature as less important for the normal operation of the binding, thus info could still be fine.
If you like, I can change it to a warning.

Reply:
The question is probably, what kind of exceptions could turn up here. Is it possible to catch more specific ones?
If the exception is something that should never happen, it should be logged as warning. If it is something that is expected to happen in certain setups (network constallations), then it can be info or even debug.

Reply:
It is mainly about degrading gracefully, there were some unexpected exceptions coming from Calimero in the past. I changed it to warning, let's see what we observe.